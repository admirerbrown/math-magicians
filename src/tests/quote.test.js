import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';

import Quotes from '../components/Quote';

describe('Test the quote component', () => {
  it('has all snapshots working', () => {
    const tree = renderer.create(<Quotes />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('has the correct text', () => {
    const { getByText } = render(<Quotes />);
    const quoteText = getByText(/Neil deGrasse Tyson, American astrophysicist and author/i);
    expect(quoteText).toBeInTheDocument();
  });

  it('renders without crashing', () => {
    render(<Quotes />);
  });
});
