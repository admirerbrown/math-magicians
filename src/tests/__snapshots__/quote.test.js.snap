// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the quote component has all snapshots working 1`] = `
<div
  className="q-div"
>
  <p
    className="q-pageText"
  >
    Somehow it’s okay for people to chuckle about not being good at math. Yet, if I said “I never learned to read,” they’d say I was an illiterate dolt. —Neil deGrasse Tyson, American astrophysicist and author
  </p>
</div>
`;
